hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java

```diff
@@ -473,8 +482,10 @@ public HttpServer2 build() throws IOException {
       HttpServer2 server = new HttpServer2(this);
 
       if (this.securityEnabled &&
-          !this.conf.get(authFilterConfigurationPrefix + "type").
-          equals(PseudoAuthenticationHandler.TYPE)) {
+          authFilterConfigurationPrefixes.stream().noneMatch(
+              prefix -> (prefix + "type")
+                  .equals(PseudoAuthenticationHandler.TYPE))
+      ) {
```

Comment:
Maybe I'm missing something but this is not doing the same as the old code block.
The old code checked if the configuration object has a key set for the prefix + type.
With the new code, you are just checking if the Stream constructed from authFilterConfigurationPrefixes does not match the prefix + type. 
Is this intentional?

Reply:
Follow-up question then: How come that this mistake was not caught by any of the tests?

Reply:
Probably there is no test for this :/

Reply:
I see. You can file a follow up jira for this if you want to :)